File: code/88265.py
def ktdxung(s):
    n=len(s)
    j=n-1
    for i in range(n//2+1):
        if s[i]!=s[j]: return False
        j-=1
    return True
s=input()
n=len(s)
stop=False
for d in range(n,0,-1):
    if stop: break
    for i in range(n-d+1):
        p=s[i:i+d]
        if ktdxung(p):
            print(p)
            stop=True

File: code/88265.cpp
#include <iostream>
#include <string>
using namespace std;
bool ktdxung(string s) {
    int n=s.length();
    int j=n-1;
    for (int i=0; i<n/2; i++) {
        if (s[i]!=s[j]) return false;
        j--;
    }
    return true;
}
int main()
{
    string s, p;
    getline(cin, s);
    int n=s.length();
    bool stop=false;
    for (int d=n; d>=1; d--) {
        if (stop) break;
        for (int i=0; i<=n-d; i++) {
            p=s.substr(i,d);
            if (ktdxung(p)) {
                cout<<p<<endl;
                stop=true;
            }
        }
    }
    return 0;
}